src/vs/base/parts/ipc/common/ipc.ts

```diff
@@ -655,12 +654,19 @@ export class ChannelClient implements IChannelClient, IDisposable {
 
 		const emitter = new Emitter<any>({
 			onWillAddFirstListener: () => {
-				uninitializedPromise = createCancelablePromise(_ => this.whenInitialized());
-				uninitializedPromise.then(() => {
-					uninitializedPromise = null;
+				const doRequest = () => {
 					this.activeRequests.add(emitter);
 					this.sendRequest(request);
-				});
+				};
+				if (this.state === State.Idle) {
+					doRequest();
+				} else {
+					uninitializedPromise = createCancelablePromise(_ => this.whenInitialized());
+					uninitializedPromise.then(() => {
+						uninitializedPromise = null;
+						doRequest();
+					});
+				}
 			},
 			onDidRemoveLastListener: () => {
```

Comment:
This block will no longer remove the listener if state is idle when the event is listened to

Reply:
Regarding your suggestion, I have a question: 
when the state is idle, it indicates that `uninitializedPromise` has completed execution and has been set to `null`. Therefore, when `onDidRemoveLastListener` occurs, it should correctly trigger `EventDispose`. Is there something I might have overlooked?

Reply:
Oh, nevermind, that should be fine